### PR TITLE
-WL Implemented a UCSBSubjects Table for the frontend

### LINKED ISSUE #14: 20-frontend-admin-menu copy
- [x] Add UCSBSubjects Table component

- [x] Possibly reusing code from previous team projects, add a UCSBSubjects table component.

- [x] Copy ACs' from previous issues.
ACs:
        - The table has columns for database fields (i.e. the attributes of the corresponding @Entity class
        - There are fixtures under /frontend/src/fixtures for this data type, similar to the other fixtures files.
        - There is a storybook entry similar to the one for UCSBDatesTable
        - 100% coverage with npm run coverage
        - 100% coverage with npx stryker run

- [x] Add to storybook, and have test coverage.

### PR DESCRIPTION
**Pull request by Wenjin09**
# Overview
Implemented a UCSBSubjects Table for the frontend and its relatively storybook and tests files.  Prepare for subsequent tasks.

In this PR, it's 100% coverage with npm run coverage and npx stryker run #14 

# Issues Addressed
Closes #14 

#Details
Moving our project's files from team03, attempt to merge these files to main:
frontend/src/fixtures/ucsbSubjectsFixtures.js
frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js
frontend/src/main/utils/UCSBSubjectUtils.js
frontend/src/stories/components/UCSBSubjects/UCSBSubjectsTable.stories.js
frontend/src/tests/components/UCSBSubjectsTest/UCSBSubjectsTable.test.js
frontend/src/tests/utils/UCSBSubjectUtils.test.js

npm run coverage:
image

Stryker:
image

### CODE REVIEW

**Comment by tai-jui** [comment]
Hey Wenjin, like Andrew just said, I think the title and description of the PR need to be more descriptive in order for it to pass staff checks.

**Comment by Wenjin09** [comment]
> Hey Wenjin, like Andrew just said, I think the title and description of the PR need to be more descriptive in order for it to pass staff checks.

**Comment by Wenjin09** [comment]
> Hey Wenjin, like Andrew just said, I think the title and description of the PR need to be more descriptive in order for it to pass staff checks.

fixed

**Review by tai-jui — DISMISSED** [review]
LGTM!

**Review by edisonnchen — CHANGES_REQUESTED** [review]
Replace the placeholder loadSubjectsTable in loadSubjectsPage to test the table

Should delete and edit be working

**Review by edisonnchen — DISMISSED** [review]
LGTM

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    //         const cell = { row: { values: { id: 17, name: "Groundhog Day" } } };
+    //         const expectedToast =  `Edit Callback called on id: 17 name: Groundhog Day`
+
+    //         // act
+    //         const result = editCallback(cell);
+
+    //         // assert
+
+    //         expect(mockToast).toHaveBeenCalledWith(expectedToast);
+
```
Comment (pconrad) [inline frontend/src/tests/utils/UCSBSubjectUtils.test.js]: Please consider removing commented out code before submitting a PR

**Review by pconrad — CHANGES_REQUESTED** [review]
Please fix title/desc and remove commented out code

**Comment by bobhe9606** [comment]
In your overview, I think you should address what changes you did and functionalities you implemented

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js (hunk 1 above)
Comment (Wenjin09) [reply frontend/src/tests/utils/UCSBSubjectUtils.test.js]: fixed

Comment (pconrad) [reply frontend/src/tests/utils/UCSBSubjectUtils.test.js]: It doesn't look fixed on my screen.  Did you push this change?

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    //         // arrange
+    //         const cell = { row: { values: { id: 17, name: "Groundhog Day" } } };
+    //         const expectedToast =  `Edit Callback called on id: 17 name: Groundhog Day`
+
+    //         // act
+    //         const result = editCallback(cell);
+
+    //         // assert
+
+    //         expect(mockToast).toHaveBeenCalledWith(expectedToast);
```
Comment (pconrad) [inline frontend/src/tests/utils/UCSBSubjectUtils.test.js]: This commented out test still need to be removed.

**Review by pconrad — CHANGES_REQUESTED** [review]

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js (hunk 2 above)
Comment (Wenjin09) [reply frontend/src/tests/utils/UCSBSubjectUtils.test.js]: forgot to push, now it should be fixed this issue.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**